### PR TITLE
Feature/initial module definition

### PR DESCRIPTION
**Pull request by pdomaleczny**
@JozefFlakus This will be refactored in the future steps.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,9 +41,12 @@
   },
   "homepage": "https://github.com/baristaJS/barista-core#readme",
   "dependencies": {
+    "@types/mkdirp": "^0.5.1",
```
Comment (JozefFlakus) [inline package.json]: move `@types` to `devDependecies`

File: package.json [hunk 2]
```diff
@@ -40,9 +41,12 @@
...
+    "mkdirp": "^0.5.1",
```
Comment (JozefFlakus) [inline package.json]: You can install additional @types definitions for this package - `@types/mkdirp`

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { Options } from './types';
```
Comment (JozefFlakus) [inline src/index.ts]: change this file to something like `interfaces` instead of `types`

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  APP_JS_TEMPLATE,
+  GITIGNORE_TEMPLATE,
+  ROUTES_INDEX_TEMPLATE,
+  API_INDEX_ROUTE_TEMPLATE,
+  README_TEMPLATE,
+  CONFIG_TEMPLATE,
+  BARISTARC_TEMPLATE,
+} from './templates';
+
+const mkdirp = require('mkdirp');
```
Comment (JozefFlakus) [inline src/index.ts]: If you will install @types for this package you can import it like any other es2015 module

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+const writeFile = require('write');
+
+function createDirectories(directories: Array<string>)  {
```
Comment (JozefFlakus) [inline src/index.ts]: you can use `string[]` instead of `Array<string>`

File: src/templates/app.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { Options } from "../types";
+
+
```
Comment (JozefFlakus) [inline src/templates/app.ts]: remove empty line

File: src/index.ts (hunk 4 above)
Comment (pdomaleczny) [reply src/index.ts]: It hasn't default export nor the name of this function as export :C

Comment (JozefFlakus) [reply src/index.ts]: Import * as mkdirp from ‚mkdirp’

**Pull request merged**